gui/wxpython/startup/locdownload.py

```diff
@@ -470,30 +564,35 @@ def __init__(self, parent, database,
         :param title: window title if the default is not appropriate
         """
         wx.Dialog.__init__(self, parent=parent, title=title)
+        cancel_button = Button(self, id=wx.ID_CANCEL)
+        self.download_button = Button(parent=self, id=wx.ID_ANY,
+                                      label=_("Do&wnload"))
+        self.download_button.SetToolTip(_("Download selected location"))
         self.panel = LocationDownloadPanel(parent=self, database=database)
-        close_button = Button(self, id=wx.ID_CLOSE)
-        # TODO: terminate download process
-        close_button.Bind(wx.EVT_BUTTON, self.OnClose)
+        cancel_button.Bind(wx.EVT_BUTTON, self.OnCancel)
+        self.Bind(wx.EVT_CLOSE, lambda evt: self.Hide())
```

Comment:
This works for mac:
`self.Bind(wx.EVT_CLOSE, self.OnCancel)` instead of `self.Bind(wx.EVT_CLOSE, lambda evt: self.Hide())`

Reply:
Yes it work but partially on wxGTK (GNU/Linux). 

1. Open location download dialog
2. Hit Download button, downloading start
3. When I hit Cancel button, `OnCancel` event handler is not trigered, and confirm modal dialog (terminate downloading) is not showed

Reply:
I’ll look into this tomorrow.